### PR TITLE
Fix error iqs in mod last

### PR DESCRIPTION
**Pull request by bartekgorny**
This PR addresses #MIM-752: errors sent in jabber:iq:last namespace were rejected with an error, which means that error response to a last activity query addressed to a subscription=none peer did not come. Also, if full jid was used, error stanzas entered an infinite loop.

This PR adds a tiny clause to the c2s function, plus tests.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7595.1](https://travis-ci.org/esl/MongooseIM/jobs/625678879) / Erlang 22.0 / small_tests / 86cb2c944e3b8d1e35bc94f0090ffe8a6208bccd
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2570/7595/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2570/7595/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-12-16_13.30.28/lib.mongooseim.logs/run.2019-12-16_13.30.29/suite.log.html)

**Comment by mongoose-im** [comment]
[7612.1](https://travis-ci.org/esl/MongooseIM/jobs/625800768) / Erlang 22.0 / small_tests / 9f222242cb6bc6d413960dd29090384034ffb57b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2570/7612/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2570/7612/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-12-16_20.50.29/lib.mongooseim.logs/run.2019-12-16_20.50.29/suite.log.html)

File: big_tests/tests/last_SUITE.erl [hunk 1]
```diff
@@ -121,6 +128,25 @@ last_server(Config) ->
...
 
+user_not_subscribed_receives_error(Config) ->
+    escalus:fresh_story(Config, [{alice, 1}, {bob, 1}], fun(Alice, Bob) ->
+        %% Alice asks about Bob's last activity
+        BobShortJID = escalus_client:short_jid(Bob),
+        escalus_client:send(Alice, escalus_stanza:last_activity(BobShortJID)),
+
+        %% server replies with an error, since there is no subscription
+        Stanza = escalus_client:wait_for_stanza(Alice),
+        escalus:assert(is_iq_error, Stanza),
```
Comment (michalwski) [inline big_tests/tests/last_SUITE.erl]: I think it'd be good to check what is the error. The assertion checks only if the iq stanza is of type error. Inside there can be sth else then we expect.

File: src/ejabberd_c2s.erl [hunk 2]
```diff
@@ -1311,9 +1311,8 @@ handle_routed_iq(From, To, Acc, StateData) ->
                        IQ :: invalid | not_iq | jlib:iq(),
                        StateData :: state()) -> routing_result().
 handle_routed_iq(From, To, Acc0, #iq{ xmlns = ?NS_LAST, type = Type }, StateData)
-  when Type /= result ->
-    %% TODO: Support for mod_last / XEP-0012. Can we move it to the respective module?
-    %%   Thanks to add_iq_handler(ejabberd_sm, ...)?
+  when Type /= result, Type /= error ->
+    % we could make iq handlers handle full jids as well, but wouldn't it be an overkill?
```
Comment (michalwski) [inline src/ejabberd_c2s.erl]: I think it would be an overkill.

**Comment by mongoose-im** [comment]
[7633.1](https://travis-ci.org/esl/MongooseIM/jobs/626808857) / Erlang 22.0 / small_tests / f15e89a96f29eaccbf0eb0481b4537ea0caf3c33
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2570/7633/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2570/7633/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-12-18_19.03.52/lib.mongooseim.logs/run.2019-12-18_19.03.52/suite.log.html)

**Pull request merged**